### DISCUSSION TITLE
CLI to apply machine patches to Machines in Maintenance mode

### DISCUSSION THREAD

**Discussion by bernardgut** [siderolabs/omni#569] [Q&A] [answered]
Hello. Dumb question but how do I programatically apply a machine config on a bunch of nodes on Omni that are still in maintenance mode ?
I tried
```
talosctl patch mc --nodes 10.2.0.8,10.2.0.9,10.2.0.10 --patch @machines/bootstrap.machinepatch.yaml
```
but I am getting:
```
rpc error: code = Unavailable desc = last connection error: connection error: desc = "transport: Error while dialing: dial tcp 10.2.0.8:50000: connect: connection refused"
```
I am assuming this is due to the fact that the machines are still in maintenance mode. Maybe the correct tool to use for this is `omnictl` in this case ? But I cant find any docs for my use-case and I need a way to create/apply the machine configs before creating the cluster without using the Omni GUI... Any ideas ?

Thanks
B.

**Comment by smira** [comment]
Machines running in maintenance mode which are adopted by Omni will not listen on their external addresses, they will only be available to Omni.

It's not clear why you need to apply machine config patches in maintenance mode, if you describe your use-case, it would be easier to provide a solution.

Omni doesn't apply user machine configuration to machines before they join the cluster, but there are other workarounds.

> Note: full machine configuration management in maintenance mode is not yet supported by Omni/Talos, so only some configuration pieces will work.

**Reply by bernardgut** [reply]
I am pretty new to Talos/Omni so maybe I misunderstood how to use the tool. The workflow is the following:

- The Omni GUI lets you create machine patches in the "machine"-><hostname>->"patches"(tab). 
- you can `omnictl sync` and `omnictl delete`, these patches are persisted regardless of the status of your nodes
- Typically I use these to set the nodes basics such as machine hostname, ip, etc.

Currently the only way I have found to create these "persisted" patches is through the Omni console. I am looking for an automated way to do it. Does that makes sense ?

**Reply by smira** [reply, marked as answer]
Right now there is no "easy" way to manage machine patches which are not bound to the cluster via the CLI.

If you create a machine config patch via Omni UI, you can find it back via `omnictl get configpatches -o yaml`, and once you found it, use that as a template, and do `omnictl apply -f` modifying the template for each machine.

**Reply by bernardgut** [reply]
Ok thanks. 

In the future it would be good to have an API for those as things like hostname/IP are very tedious to do in the GUI
